src/Prettus/Repository/Eloquent/BaseRepository.php

```diff
@@ -753,6 +753,35 @@ public function delete($id)
         return $deleted;
     }
 
+    /**
+     * Force a hard delete on a soft deleted model to a entity in repository by id
+     *
+     * @param $id
+     *
+     * @return int
+     */
+    public function forceDelete($id)
+    {
+        $this->applyScope();
+
+        $temporarySkipPresenter = $this->skipPresenter;
+        $this->skipPresenter(true);
+
+        $model = $this->find($id);
+        $originalModel = clone $model;
+
+        $this->skipPresenter($temporarySkipPresenter);
+        $this->resetModel();
+
+        event(new RepositoryEntityDeleting($this, $model));
```

Comment:
I think you need to create a new class for force deleting and force deleted.

Reply:
Yes I agree.
But I would just like to add this new event and not remove the delete events, because the force delete is also an delete and must trigger this event.

Reply:
> because the force delete is also an delete and must trigger this event.

So it require to check if its `->trashed()` in deleting or deleted listeners?

Reply:
@lloricode I saw in Eloquent that the exclusion events are the same. But they are separate, as you suggested.
So, I'm going to do an event for ForceDeleting and ForceDeleted.
https://github.com/illuminate/database/blob/7.x/Eloquent/SoftDeletes.php#L52